drivers/source/SerialBase.cpp

```diff
@@ -272,6 +272,7 @@ SerialBase::~SerialBase()
     for (int irq = 0; irq < IrqCnt; irq++) {
         attach(NULL, (IrqType)irq);
     }
+    serial_free(&_serial);
```

Comment:
#10924 has just being merged, and this change logically conflicts with that. You'll need to rebase and look again at the updated `SerialBase`.

After #10924, I believe this `serial_free` must be made conditional (on `_rx_enabled || _tx_enabled`, I think), and should be a call to `_deinit`.

Reply:
Please rebase, this PR should contain just nrf52 fix - should be sufficient